gmail/snippets/settings_snippet/settings_snippets.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+import os
+from datetime import datetime, timedelta
+from googleapiclient.discovery import build
+from google_auth_oauthlib.flow import InstalledAppFlow
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from googleapiclient.errors import HttpError
+from numpy import long
+
+# If modifying these scopes, delete the file token.json.
+
+SCOPES = ['https://www.googleapis.com/auth/gmail.settings.basic',
+          'https://www.googleapis.com/auth/gmail.settings.sharing']
+
+
+def main():
+    """Shows basic usage settings snippets of gmail API.
+        """
+    creds = None
+    # The file token.json stores the user's access and refresh tokens, and is
+    # created automatically when the authorization flow completes for the first
+    # time.
+    if os.path.exists('token.json'):
+        creds = Credentials.from_authorized_user_file('token.json', SCOPES)
+    # If there are no (valid) credentials available, let the user log in.
+    if not creds or not creds.valid:
+        if creds and creds.expired and creds.refresh_token:
+            creds.refresh(Request())
+        else:
+            flow = InstalledAppFlow.from_client_secrets_file('credentials.json', SCOPES)
+            creds = flow.run_local_server(port=0)
+        # Save the credentials for the next run
+        with open('token.json', 'w') as token:
+            token.write(creds.to_json())
+
+    try:
+        # Gmail API service instance
+        service = build('gmail', 'v1', credentials=creds)
+
+        # Call update_signature function
+        update_signature(service=service)
+        # Call create_filter function
+        create_filter(service=service)
+        # Call enable_forwarding function
+        enable_forwarding(service=service, real_forwarding_address='gduser2@workspacesamples.dev')
```

Comment:
Per previous conversation, this looks like a case where trying to combine things into one file doesn't help all that much. The snippets still end up lacking context and the main method calling all the functions indiscriminately is potentially confusing (e.g. not clear at first look if these are independent of one another.)

Same for send_mail.py